### PR TITLE
Use cpuid intrinsics instead of asm code

### PR DESCRIPTION
**Pull request by sipa**
Less platform-specific code is better.

### CODE REVIEW

**Comment by laanwj** [comment]
> Let's see if this works on all platforms...

It builds on OpenBSD at least! Keeping #10670 in mind that probably means it will work on anything.

File: src/random.cpp [hunk 1]
```diff
@@ -72,17 +76,8 @@ static bool rdrand_supported = false;
...
-    uint32_t tmp;
-    __asm__ ("mov %%ebx, %1; cpuid; mov %1, %%ebx": "=a"(eax), "=g"(tmp), "=c"(ecx), "=d"(edx) : "a"(1));
-#else
-    uint32_t ebx;
-    __asm__ ("cpuid": "=a"(eax), "=b"(ebx), "=c"(ecx), "=d"(edx) : "a"(1));
-#endif
-    //! When calling cpuid function #1, ecx register will have this set if RDRAND is available.
-    if (ecx & CPUID_F1_ECX_RDRAND) {
+    uint32_t eax, ebx, ecx, edx;
+    if (__get_cpuid(1, &eax, &ebx, &ecx, &edx) && (ecx & CPUID_F1_ECX_RDRAND)) {
```
Comment (laanwj) [inline src/random.cpp]: Definitely an improvement...

**Comment by TheBlueMatt** [comment]
ACK 674848fe1c43fb88870cf5ba16fca4e2524da793 (I didnt actually test on other platforms, so I'm just kinda assuming they support this stuff).

**Review by gmaxwell — APPROVED** [review]
ACK.

**Pull request merged**